Implemented Transactions in dbUpdateTable

Tim113 opened the pull request:
The Update Table funtion was not using transactons when deleateing and then updateing records in the db.  This could have lead to the row being delated and then lost if the connection was lost durin the function.  To test is use the code below, killing the connection to the db between the deleat and the write.

```
### Use DBI transcation so that the row is deleated iff the row is then re-inserted
  DBI::dbWithTransaction(
    conn = con,
    code = {
      dbDeleteRowByKey(con, name, dt)
      if(verbose) message("Deleating row(s) form database")

      Sys.sleep(120) # Time to kill the connection

      RMySQL::dbWriteTable(con, name, dt, row.names = FALSE, append = TRUE)
      if(verbose) message("Writing row(s) to database")
    }
  )
```

Also changed the verbose messages to make them even more verboser.

Commit:
Implemented Transactions in dbUpdateTable

The Update Table funtion was not using transactons when deleateing and then updateing records in the db.  This could have lead to the row being delated and then lost if the connection was lost durin the function.  To test is use the code below, killing the connection to the db between the deleat and the write.

```
### Use DBI transcation so that the row is deleated iff the row is then re-inserted
  DBI::dbWithTransaction(
    conn = con,
    code = {
      dbDeleteRowByKey(con, name, dt)
      if(verbose) message("Deleating row(s) form database")

      Sys.sleep(120) # Time to kill the connection

      RMySQL::dbWriteTable(con, name, dt, row.names = FALSE, append = TRUE)
      if(verbose) message("Writing row(s) to database")
    }
  )
```

Also changed the verbose messages to make them even more verboser.

AkhilNairAmey commented:
Can you write 2 quick test cases:
1) shows the delete failed 
2) shows the delete succeeds with DBI

Also is there not a better way to kill the connection?
`rm(con); gc ()`?

AkhilNairAmey commented:
Good catch tho

AkhilNairAmey commented:
And by 2) i mean just 1) because i just understood the point. Just need a test case to confirm it catches the transaction

Tim113 commented:
By Test cases do you mean with `testthat`, or an example of it working like in the readme, and yes `rm(con); gc ()` is clearly a better way to kill the connection.

yupimaki commented:
Yeah with testthat in the test directory so that

On Tue, 4 Apr 2017, 10:01 Tim Jones, <notifications@github.com> wrote:

> By Test cases do you mean with testthat, or an example of it working like
> in the readme, and yes rm(con); gc () is clearly a better way to kill the
> connection.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AkhilNairAmey/dbUpdateTable/pull/7#issuecomment-291437973>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFACEMaUZeUg0JxBTEZOlHbB0bmdko1Iks5rsgbagaJpZM4Myj9J>
> .
>

Commit:
Tests Created for dbUpdateTable

These tests test the regular functions of dbUpdate Table and its ability to use transactions to reset a partal change to the db if the connection is lost.

This involoved including an argument in dbUpdateTable to kill the connection to force it to break.

Commit:
Fix in the name of the transaction text

Commit:
Corrected Documentation for the package

The dependency’s for the package have been changed to specify the version of DBI that is required for the transaction functionality
